components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/group/SCIMGroupHandlerTest.java

```diff
@@ -0,0 +1,305 @@
+import static org.mockito.MockitoAnnotations.initMocks;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+@PrepareForTest({IdentityDatabaseUtil.class,StringUtils.class,SCIMCommonUtils.class,SCIMGroupHandler.class})
+@PowerMockIgnore("java.sql.*")
+public class SCIMGroupHandlerTest {
+
+    @Mock
+    private GroupDAO mockedGroupDAO;
+
+    @Mock
+    private Connection connection;
+
+    @Mock
+    private PreparedStatement mockedPreparedStatement;
+
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        initMocks(this);
+
+    }
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void testAddMandatoryAttributes() throws Exception {
+
+        ResultSet resultSet = mock(ResultSet.class);
+        mockStatic(SCIMCommonUtils.class);
+        mockStatic(IdentityDatabaseUtil.class);
+
+        when(SCIMCommonUtils.getSCIMGroupURL(anyString())).thenReturn("ID");
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(false);
+        when(mockedGroupDAO.isExistingGroup(SCIMCommonUtils.getGroupNameWithDomain("GROUP_NAME"),1)).thenReturn(false);
+
+        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
+        scimGroupHandler.addMandatoryAttributes(anyString());
+
+    }
+
+    @Test
+    public void testGetGroupAttributesByName() throws Exception {
+        Assert.assertNull(new SCIMGroupHandler(1).getGroupAttributesByName("managers"));
+    }
+
+    @Test
+    public void testGetGroupAttributesById() throws Exception {
+        Assert.assertNull(new SCIMGroupHandler(1).getGroupAttributesById("1"));
+    }
+
+    @Test
+    public void testCreateSCIMAttributes() throws Exception {
+        ResultSet resultSet = mock(ResultSet.class);
+
+        mockStatic(IdentityDatabaseUtil.class);
+        mockStatic(SCIMCommonUtils.class);
+
+        Group group = new Group();
+
+        Date date = new Date();
+        group.setCreatedDate(date);
+        group.setLastModified(date);
+        group.setLocation("LOCATION_URI");
+
+        when(IdentityDatabaseUtil.getDBConnection()).thenReturn(connection);
+        when(connection.prepareStatement(anyString())).thenReturn(mockedPreparedStatement);
+        when(mockedPreparedStatement.executeQuery()).thenReturn(resultSet);
+        when(resultSet.next()).thenReturn(false);
+        when(mockedGroupDAO.isExistingGroup(SCIMCommonUtils.getGroupNameWithDomain("NON_EXISTANT_GROUP_NAME"),1)).thenReturn(false);
+
+        SCIMGroupHandler scimGroupHandler = new SCIMGroupHandler(1);
+        scimGroupHandler.createSCIMAttributes(group);
+
```

Comment:
always assert some conditions/behaviors after the test.

Reply:
done